File: src/01_introduction.R

# 1. packages -------------------------------------

library(tidyverse)
library(lme4)

# 1. fit: linear model ----------------------------
lm_model = lm(Reaction ~ Days, data = sleepstudy)
summary(lm_model)

# 2. fit: mixed model -----------------------------

mixed_model = lmer(Reaction ~ Days + (1 + Days|Subject), data = sleepstudy)
summary(mixed_model)

print(VarCorr(mixed_model), comp=c('Var', 'Std')) 

# random effects
random_effects = ranef(mixed_model)$Subject
random_effects

random_coefficients = coef(mixed_model)$Subject

# fixed effects
fixed_effects = fixef(mixed_model)
random_effects  = 
  random_coefficients %>% 
  mutate(Subject = factor(unique(sleepstudy$Subject)))

sleepstudy %>% 
  ggplot(aes(Days, Reaction)) +
  geom_point(aes(color = Subject), alpha = .25) +
  geom_abline(
    aes(
      intercept = fixed_effects['(Intercept)'],
      slope = fixed_effects['Days']),
    color = 'darkred',
    size = 2) +
  geom_abline(
    aes(
      intercept = `(Intercept)`, 
      slope = Days,
      color = Subject),
    size = .5,
    alpha = .25,
    data = random_effects) 

File: src/01_introduction.jl

# installing packages
import Pkg
Pkg.add("CSV")
Pkg.add("DataFrames")
Pkg.add("MixedModels")
Pkg.add("TableView")

# load packages
using CSV
using DataFrames
using MixedModels
using TableView

# read csv
df = DataFrame(CSV.File("./data/sleepstudy.csv"))

# inspect df
first(df, 5)
showtable(df)
describe(df)

# change column type
transform!(df, :Subject => categorical, renamecols=false)

# fit mixed model
mixed_model = fit(MixedModel, @formula(Reaction ~ Days + (Days|Subject)), df, REML=true)
